lerna: repackage with buildNpmPackage

Thao-Tran opened the pull request:
Repackage `nodePackages.lerna` to be built using `buildNpmPackage` and create variants for `lerna_6` and `lerna_8`.

Refs: #229475


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
lerna: init at 8.1.9

* Include lerna_6 and lerna_8 variants

Commit:
maintainers: add ThaoTranLePhuong

Thao-Tran commented on pkgs/development/tools/lerna/package-lock.v6.json:
Unfortunately needed to include the package-lock files because the one to the GitHub repo isn't compatible with cached installs.

SuperSandro2000 commented:
Since we now commit the lockfile, do we really need to keep 2 versions? is version 8 not enough?

Thao-Tran commented:
@SuperSandro2000  I added a v6 variant because it's the version that I need to use for my company's codebase since v8 introduced a lot of breaking changes, but I understand if having two lockfiles is too much. Let me know if you have a strong opposition to maintaining both versions and I can just keep it in my nixpkgs fork instead.

Hmm I am wondering now though, would it be considered bad practice if I changed this to point to a personal fork of `lerna` that actually provides a proper lockfile for both versions, so we don't have to keep the lockfile in the `nixpkgs` VCS?

SuperSandro2000 commented:
I am just going to assume that this is a general thing and we merge this for now :)

> Hmm I am wondering now though, would it be considered bad practice if I changed this to point to a personal fork of `lerna` that actually provides a proper lockfile for both versions, so we don't have to keep the lockfile in the `nixpkgs` VCS?

That makes it harder for everyone to update.
